query-big-database/solution.md

```diff
@@ -146,8 +210,16 @@ Pre-optimization: 9173.750 ms
 Post-optimization: 2794.690 m
 
 ### 3. Optimize Partition
+
 ### 4. Database tuning
 
+SQL query which we execute, has to load the data into memory to perform any operation. It has to hold the data in memory, till the execution completes.
+
+If the memory is full, data will be spilled to disk which causes the transactions to run slow, because disk I/O is time taking task.
+
+So, a good amount of memory enough to fit the data that is being processes as part of SQL is needed.
+
+So you need increase server resource as much as you can base on your scale and acceptable cost to prevent out of memory or out of CPU utilization and increase query process speed
```

Comment:
Can you give us some advice for config how much memory and CPU

Reply:
> Can you give us some advice for config how much memory and CPU

Thx you, I updated